pulsar/consumer_partition.go

```diff
@@ -478,13 +482,36 @@ func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, header
 		return err
 	}
 
-	uncompressedHeadersAndPayload, err := pc.Decompress(msgMeta, headersAndPayload)
+	var uncompressedHeadersAndPayload internal.Buffer
+
+	// decrypt the data if needed
+	if msgMeta.EncryptionParam != nil {
+		if pc.options.messageCrypto != nil && (pc.options.cryptoKeyReader != nil || pc.options.dataKeyCrypto != nil) {
+			if pc.options.dataKeyCrypto != nil {
```

Comment:
https://github.com/apache/pulsar/wiki/PIP-4:-Pulsar-End-to-End-Encryption#consumer-1

Should respect config for ConsumerCryptoFailureAction, with proper handling for when the message contains a batch of messages.